code/datums/elements/reagent_refiller.dm

```diff
@@ -0,0 +1,60 @@
+///Reagent refiller, refills any drinks poured out of the reagent container (and is allowed within the whitelisted reagents).
+/datum/element/reagent_refiller
+	element_flags = ELEMENT_DETACH
+
+	///Time to refill
+	var/time_to_refill
+	///Cell to use power of
+	var/datum/weakref/cell_to_use
```

Comment:
You can't store a reference like this within an element.
Because elements are shared between all datums they are attached to, holding a reference to a cell will mean all instances of the element in existence will draw from the same cell.

Convert this to a component, instead

Reply:
oh man ok